ghost/core/core/server/services/collections/BookshelfCollectionsRepository.js

```diff
@@ -122,19 +132,68 @@ module.exports = class BookshelfCollectionsRepository {
             {id: data.id},
             {
                 require: false,
-                transacting: options.transaction
+                transacting: options.transaction,
+                withRelated: ['collectionPosts']
             }
         );
 
         if (!existing) {
             await this.#model.add(data, {
                 transacting: options.transaction
             });
+            const thingsToInsert = collection.posts.map((postId, index) => {
+                return {
+                    id: (new ObjectID).toHexString(),
+                    sort_order: collection.type === 'manual' ? index : 0,
+                    collection_id: collection.id,
+                    post_id: postId
+                };
+            });
+            if (thingsToInsert.length > 0) {
```

Comment:
I'd go as far as move this block higher and check for `if (collection.posts.length > 0)`

Reply:
Coming from a place where reading a code would be like: oh do this whole piece of logic if there's anything to it and mapping is part of this logic imo.

Reply:
The checks are around the calls to the database because those will throw if passed empty arrays. Else where we've filtered the arrays so I want to keep the check close to where they're used to avoid trying to insert 0 values